modules/twinkleprod.js

```diff
@@ -172,6 +172,58 @@ Twinkle.prod.callback.prodtypechanged = function(event) {
 };
 
 Twinkle.prod.callbacks = {
+	checkpriors: function(apiobj) {
+		var xmlDoc = apiobj.responseXML;
+		var statelem = apiobj.statelem;
+		var params = apiobj.params;
+
+		// Check existence of AfD, would be easier if mw.Title.exists() worked [[phab:T184953]]
+		var exists;
+		if (namespace === 'article') {
+			var query = {
+				'action': 'query',
+				'prop': 'revisions',
+				'titles': 'Wikipedia:Articles for deletion/' + Morebits.pageNameNorm
+			};
+			var wikipedia_api = new Morebits.wiki.api('Checking for prior AfDs', query, function(apiobj) {
+				var xmlDoc = apiobj.responseXML;
+				exists = $(xmlDoc).find('rev').attr('revid');
+			});
+			// Wait for API call to finish
+			wikipedia_api.post({
```

Comment:
`async` used so we can avoid creating separate functions for mainspace and as well as not showing "Checking for AfDs" for filespace PRODs